src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -554,22 +554,37 @@ public function serve_request( $path = null ) {
 	 *
 	 * @since 4.4.0
 	 * @since 5.4.0 The `$embed` parameter can now contain a list of link relations to include.
+	 * @since 6.3.0 The `$request` parameter was added.
 	 *
 	 * @param WP_REST_Response $response Response object.
 	 * @param bool|string[]    $embed    Whether to embed all links, a filtered list of link relations, or no links.
+	 * @param WP_REST_Request  $request  Request object.
 	 * @return array {
 	 *     Data with sub-requests embedded.
 	 *
 	 *     @type array $_links    Links.
 	 *     @type array $_embedded Embedded objects.
 	 * }
 	 */
-	public function response_to_data( $response, $embed ) {
+	public function response_to_data( $response, $embed, $request = null ) {
 		$data  = $response->get_data();
 		$links = self::get_compact_response_links( $response );
```

Comment:
Is it better to check that links are needed in the first place before getting them?

Reply:
Agreed, it would be more efficient to skip getting the links at all if we know they are not requested. That would avoid wasted effort generating a links object we don't go on to use.

Reply:
Right, it's respecting the `_fields` in function `rest_filter_response_fields` but

```
$links = self::get_compact_response_links( $response );
if ( ! empty( $links ) ) {
	$data['_links'] = $links;
}
```

 This code will add the _links anyway, it's not validating the `_fields`. so I decide to add the $request object as parameter.